crates/snapbox/src/data.rs

```diff
@@ -102,6 +117,8 @@ impl Data {
         match self.inner {
             DataInner::Binary(data) => Self::binary(data),
             DataInner::Text(data) => Self::text(op(&data)),
+            #[cfg(feature = "json")]
+            DataInner::Json(json) => Self::json(json),
```

Comment:
`map_text` is meant to perform normalizations, whether newline or match-syntax.  We'll need a more general solution when adding structured data.

One idea is a `normalize` function that accepts a policy.  We could provide users with a `Newline` policy or a `Matches` policy.  This won't let users write their own normalization routines but that might be ok for now.

Reply:
I figured that a change like this would have to happen. I will try and figure something out